File: src/main/java/com/example/library_management_system/service/BookService.java
package com.example.library_management_system.service;

import com.example.library_management_system.entity.Book;
import com.example.library_management_system.repository.AuthorRepository;
import com.example.library_management_system.repository.BookRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class BookService {

    @Autowired
    private BookRepository bookRepository;

    @Autowired
    private final AuthorRepository authorRepository;

    public BookService(BookRepository bookRepository, AuthorRepository authorRepository) {
        this.bookRepository = bookRepository;
        this.authorRepository = authorRepository;
    }

    public Book addBook(Book book) {
        return bookRepository.save(book);
    }

    public List<Book> getAllBooks() {
        return bookRepository.findAll();
    }

    public Book getBookById(Long id) {
        return bookRepository.findById(id).orElse(null);
    }

    public List<Book> getBooksByAuthorId(Long authorId) {
        return bookRepository.findByAuthorId(authorId);
    }

    public Book updateBook(Long id, Book bookDetails) {
        Book book = bookRepository.findById(id).orElse(null);
        if (book != null) {
            book.setTitle(bookDetails.getTitle());
            book.setIsbn(bookDetails.getIsbn());
            book.setAuthorId(bookDetails.getAuthorId());
            return bookRepository.save(book);
        }
        return null;
    }

    public void deleteBook(Long id) {
        bookRepository.deleteById(id);
    }
}

File: app/src/services/BookService.js
import axios from 'axios';

const baseURL = 'http://localhost:8080/api/v1';

const booksURL = baseURL + '/books';


const BookService = {
  getAllBooks: () => axios.get(booksURL),
  getBookById: (id) => axios.get(`${booksURL}/${id}`),
  addBook: (book) => axios.post(booksURL, book),
  updateBook: (id, book) => axios.put(`${booksURL}/${id}`, book),
  deleteBook: (id) => axios.delete(`${booksURL}/${id}`),
  getAllBooksByAuthorId: (id) => axios.get(`${baseURL}/authors/${id}/books`),
};

export default BookService;
